andela/activo-api
# #157767387 Add dynamic filter to support entity query

#### What does this PR do?
- Enable dynamic filtering of entity query result
#### Description of Task to be completed?
- Define a class that dynamically filters a query result
- Refactor the `_query` method of `model_operations` class to return filtered result
#### How should this be manually tested?
- Create a branch `ch-add-dynamic-filter-157767387` and pull from its remote counterpart
- Activate the python environment by running `venv/bin/activate`
- Run `pip install -r requirements.txt` to install all dependencies
- In your terminal run `flask run` to start the application.
- Use Postman, make a get request to the URL `http://127.0.0.1:5000/api/v1/asset-categories/stats`. This URL has no filter and should return all the asset categories in the database.
- Make filtered get requests to the URL in the format: `http://127.0.0.1:5000/api/v1/asset-categories/stats?where=x,y,z`. **x** is an attribute, such as `name`, of the AssetCategory class. **y** is comparator and **z** is the string value to compare.
- Possible values of **y** are: **eq**(equal to); **ne**(not equal to); **lt**(less than); **gt**(greater than); **le**(less than or equal to); **ge**(greater than or equal to); **like**(like). 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
 - ch-add-dynamic-filter-to-support-query-operation-157767387
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/24465059/41229922-b670cece-6d75-11e8-8e2b-fc3db5ef5687.png)

![image](https://user-images.githubusercontent.com/24465059/41229925-bf081420-6d75-11e8-9cbe-4643f6d05c79.png)

![image](https://user-images.githubusercontent.com/24465059/41229933-c6ea8c90-6d75-11e8-965d-92e7cc21d7e5.png)


#### Questions:
N/A
